### DISCUSSION TITLE
Maximum load generating capacity

### DISCUSSION THREAD

**Discussion by fvoellma** [artilleryio/artillery#1842] [Artillery]
New to Artillery, trying to understand pros/cons vs. Gatling (also running in AWS). 

Can anyone point me to studies showing the performance of Artillery's different versions?

Thanks in advance.

**Comment by hassy** [comment]
Hi @fvoellma 👋 it's a real "it depends" kind of question. Can you tell us more about your requirements? (what sort of a load, from how many clients, protocol, type of service etc)

As a general answer, Artillery is designed to make it super-easy to scale out and run a distributed load test from many worker nodes. You can use AWS Lambda to scale out to hundreds of thousands or millions of RPS without setting up or managing any infrastructure.

This blog post shows a load test running at 100k RPS: https://www.artillery.io/blog/this-load-test-cost-us-how-much

This recent talk by Solutions Architects from AWS at Serverless Summit shows another Artillery test running at 100k RPS - https://www.youtube.com/watch?v=TFHF_PttO9g

We've done internal tests at over 1M RPS. Really the constraint is AWS Lambda concurrency limit in your AWS account. :)

**Reply by fvoellma** [reply]
Hi @hassy. Thanks for the quick response. I'll look at the provided links.
I'm looking at Artillery for possible use in many services, each with differing requirements. 10's kRPS is probable near term maximum. Good to know it's been tested far beyond that!
